ansible/playbook.yml

```diff
@@ -0,0 +1,72 @@
+---
+- name: Sanghaplanner Provisioning
+  hosts: all
+  become: yes
+
+  tasks:
+  - name: add php repository
+    apt_repository: repo='ppa:ondrej/php'
+
+  - name: update packages
+    apt: update_cache=yes
+
+  - name: install packages
+    apt: name={{ item }} state=latest
+    with_items:
+      - vim
+      - mysql-server
+      - python-mysqldb
+      - php5.6
+      - php5.6-fpm
+      - php5.6-mbstring
+      - php5.6-dom
+      - php5.6-mcrypt
+      - php5.6-mysql
+      - nginx
+      - nodejs-legacy
+      - npm
+
+  - name: unlink default nginx configuration
+    file: path=/etc/nginx/sites-enabled/default state=absent
+
+  - name: delete default nginx configuration
+    file: path=/etc/nginx/sites-available/default state=absent
+
+  - name: copy nginx sanghaplanner configuration
+    template: src=templates/nginx.cfg.j2 dest=/etc/nginx/sites-available/sanghaplanner
+
+  - name: enable nginx sanghaplanner configuration
+    file: src=/etc/nginx/sites-available/sanghaplanner dest=/etc/nginx/sites-enabled/sanghaplanner state=link
+
+  - name: create database
+    mysql_db: name=sanghaplanner state=present
+
+  - name: create database user
+    mysql_user: name=vagrant password=vagrant host=% priv=*.*:ALL state=present
+
+  - name: allow remote mysql connections
+    lineinfile: dest=/etc/mysql/my.cnf regexp='bind-address\s*=\s*127\.0\.0\.1\s*' line='bind-address = 0.0.0.0' state=present
+
+  - name: run database migrations
+    shell: cd /vagrant && php artisan migrate
+
+  - name: download composer
+    shell: curl -sS https://getcomposer.org/installer | sudo php -- --install-dir=/usr/local/bin --filename=composer
+
+  - name: composer install
+    shell: cd /vagrant && composer install
+
+  - name: npm prune
+    shell: cd /vagrant && npm prune
+
+  - name: disable npm progress bar
+    shell: npm set progress=false
+
+  - name: npm install
+    shell: cd /vagrant && npm install
+
+#  - name: download composer
+#    get_url:
+#      url: https://getcomposer.org/installer
+#      dest: /usr/local/bin/composer
+#      mode: 0755
```

Comment:
This works, but you get a file with the content of composer-setup.php at the destination.
You could use a dest of **/tmp/composer-setup.php** and then make another task which installs it to be /usr/local/bin/composer